### ISSUE TITLE
Can't use `jupyter server extension <anything>` if extension is not importable.

### ISSUE THREAD

**Issue by Carreau** [jupyter-server/jupyter_server#784]
- install extension with arbitrary name that you develop
- uninstall or break it. 
- try to use jupyter server extension <list|disable|anything>

You can't because it fails at 

```
class ExtensionPackage(HasTraits):
    """An API for interfacing with a Jupyter Server extension package.

    Usage:

    ext_name = "my_extensions"
    extpkg = ExtensionPackage(name=ext_name)
    """

    name = Unicode(help="Name of the an importable Python package.")
    enabled = Bool(False).tag(config=True)

    def __init__(self, *args, **kwargs):
        # Store extension points that have been linked.
        self._linked_points = {}
        super().__init__(*args, **kwargs)

    _linked_points = {}

    @validate_trait("name")
    def _validate_name(self, proposed):
        name = proposed["value"]
        self._extension_points = {}
        try:
            self._module, self._metadata = get_metadata(name)
        except ImportError:
            pass
            raise ExtensionModuleNotFound(
                "The module '{name}' could not be found. Are you "
                "sure the extension is installed?".format(name=name)
            )
        # Create extension point interfaces for each extension path.
        for m in self._metadata:
            point = ExtensionPoint(metadata=m)
            self._extension_points[point.name] = point
        return name

```

2) ImportError is not the same as ModuleNotFound,  it should catch module not found an importError separately.
3) there are several instance of `self.serverapp.reraise_server_extension_failures`, 
though `self.serverapp` is `None` for me, so this fails everywhere.

**Comment by vidartf** [comment]
I think #887 addressed point 3 above (duplicate of #676 ?), but note sure about point 2?

**Comment by blink1073** [comment]
This is no longer the case, I just made an extension raise an error on import, and I get:

```bash
jupyter server extension list
Config dir: /Users/silvester/.jupyter

Config dir: /Users/silvester/workspace/.venvs/jupyter_server_terminals/etc/jupyter
    jupyter_server_terminals enabled
    - Validating jupyter_server_terminals...
       X hahaha

Config dir: /usr/local/etc/jupyter
```
